packages/block-editor/src/components/block-draggable/index.js

```diff
@@ -61,6 +78,55 @@ const BlockDraggable = ( {
 		};
 	}, [] );
 
+	// Find the root of the editor iframe.
+	const blockRef = useBlockRef( clientIds[ 0 ] );
+	const editorRoot = blockRef.current?.closest( 'body' );
+
+	// Add a dragover event listener to the editor root to track the blocks being dragged over.
+	// The listener has to be inside the editor iframe otherwise the target isn't accessible.
+	// Check if the dragged blocks are allowed inside the target. If not, grey out the draggable.
+	useEffect( () => {
+		if ( ! editorRoot ) {
+			return;
+		}
+
+		const onDragOver = ( event ) => {
+			if ( ! event.target.closest( '[data-block]' ) ) {
+				return;
+			}
+
+			const newTargetClientId = event.target
+				.closest( '[data-block]' )
+				.getAttribute( 'data-block' );
+			//Only update targetClientId if it has changed
+			// and if it's a container block
+			if (
+				targetClientId !== newTargetClientId &&
+				getBlockListSettings( newTargetClientId )
+			) {
+				setTargetClientId( newTargetClientId );
+			}
+			// Update the body class to reflect if drop target is valid.
+			// This has to be done on the document body because the draggable
+			// chip is rendered outside of the editor iframe.
+			if ( ! isDropTargetValid && ! visibleInserter ) {
+				window?.document?.body?.classList?.add(
+					'block-draggable-invalid-drag-token'
+				);
+			} else {
+				window?.document?.body?.classList?.remove(
+					'block-draggable-invalid-drag-token'
+				);
+			}
+		};
```

Comment:
If we wanted to avoid attaching classes to the body element, would it work to pass an `isDisabled` boolean to `BlockDraggableChip` and have that component output an extra classname alongside `block-editor-block-draggable-chip-wrapper` if that flag is set? Or, alternately, update `BlockDraggableChip` to accept a `className` param?

Reply:
I tried all of that already 😄 

The problem is that Draggable makes a copy of the chip in its [start function](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/draggable/index.tsx#L100C2-L100C2) and that stored copy is what gets moved around, so its state never changes from the initial render. That's why I'm using the body classname to update the value of a CSS custom property: it was the simplest way I found to affect the copy of the draggable chip in real time. I also wanted to avoid any changes to Draggable itself, because it's a generic component that shouldn't have to know anything about what it's dragging.

Fwiw Draggable adds a class to body while dragging in order to change the cursor, which is what gave me the idea in the first place.

Reply:
Ah, of course. Sounds good to me! Thanks for the explanation 🙂